src/lib/market-service/zerion/zerion.ts

```diff
@@ -0,0 +1,53 @@
+import { type AssetId, fromAssetId } from '@shapeshiftoss/caip'
+import type {
+  FindAllMarketArgs,
+  HistoryData,
+  MarketCapResult,
+  MarketData,
+  MarketDataArgs,
+  PriceHistoryArgs,
+} from '@shapeshiftoss/types'
+import Axios from 'axios'
+import { setupCache } from 'axios-cache-interceptor'
+
+import type { MarketService } from '../api'
+import { DEFAULT_CACHE_TTL_MS } from '../config'
+import type { ZerionFungibles, ZerionMarketData } from './types'
+import { zerionMarketDataToMarketData } from './utils'
+
+const axios = setupCache(Axios.create(), { ttl: DEFAULT_CACHE_TTL_MS, cacheTakeover: false })
+
+export class ZerionMarketService implements MarketService {
+  baseUrl = 'https://zerion.shapeshift.com'
+
+  getZerionTokenMarketData = async (assetId: AssetId): Promise<ZerionMarketData | undefined> => {
+    const { assetReference } = fromAssetId(assetId)
+    const url = `${this.baseUrl}/fungibles/${assetReference}`
+    const { data: res } = await axios.get<ZerionFungibles>(url)
+    return res.data.attributes.market_data
+  }
+
+  async findAll(_args?: FindAllMarketArgs): Promise<MarketCapResult> {
+    // TODO - unimplemented
```

Comment:
For a follow-up PR, less important than getting custom asset market data ready before custom token marketing.

Reply:
It'll stay last for now, but we should absolutely protect against it being moved! Excellent thought.

Addressed in https://github.com/shapeshift/web/pull/7489/commits/584a9e383855aa0792b5deeac1a1b8bc71701d58

Monkey patched to be the first provider, and it gracefully warns in the console and then falls back to the next provider:

<img width="1187" alt="Screenshot 2024-08-06 at 9 16 23 PM" src="https://github.com/user-attachments/assets/a386dfea-5bba-44c2-82e2-21345fefd882">